### PR TITLE
Add info button item data

### PR DESCRIPTION
**Pull request by nickoola**

### CODE REVIEW

File: src/_utils/item/Item.tsx [hunk 1]
```diff
@@ -18,6 +20,8 @@ export interface ItemProps {
   readonly highlighted?: boolean
   readonly isClickable?: boolean
   readonly leftTitle?: string
+  readonly isLeftTitleInfo?: boolean
+  readonly leftTitleInfoIcon?: string
```
Comment (nickoola) [inline src/_utils/item/Item.tsx]: leftTitleInfoIcon >> leftTitleIconInfo

**Comment by nickoola** [comment]
Capture d’écran 2020-01-05 à 11 15 59
[CHROME RENDER]
[NEED TO BE TESTED ON OTHER BROWSER]

File: src/_utils/item/Item.tsx (hunk 1 above)
Comment (mcellier) [reply src/_utils/item/Item.tsx]: That's a lot of props you add here. Maybe we can go down to 2 (naming proposal, do what you want with this)
- infoIconLabel
- infoIconCallback 
We won't (i hope) have these icons elsewhere, do the `left` part may be ... left aside.

The `isLeftTitleInfo` seems to be unnecesserrary: you can check either the label or the function

File: src/_utils/item/Item.tsx [hunk 2]
```diff
@@ -108,12 +116,23 @@ const Item = ({
       <div className="kirk-item-leftText">
         {leftTitle && (
           <Text
-            className={leftBody ? 'kirk-item-title' : null}
+            className={isLeftTitleInfo ? 'kirk-item-title--withInfo' : null}
```
Comment (mcellier) [inline src/_utils/item/Item.tsx]: no more `kirk-item-title`?

File: src/itemData/index.tsx [hunk 3]
```diff
@@ -8,6 +8,9 @@ interface ItemDataProps {
   readonly mainInfo: string
   readonly className?: Classcat.Class
   readonly mainTitle?: string
+  readonly isMainTitleInfo?: boolean
+  readonly mainTitleInfoIcon?: string
+  readonly onClickInfo?: (event: React.MouseEvent<HTMLElement>) => void
```
Comment (mcellier) [inline src/itemData/index.tsx]: React.MouseEvent on touch?

File: src/_utils/item/Item.tsx (hunk 2 above)
Comment (nickoola) [reply src/_utils/item/Item.tsx]: I think it's a legacy rule. There is no logic display link between `leftBody` inside `leftTitle`.
And `kirk-item-title` doesn't apply any styles.

File: src/itemData/index.tsx (hunk 3 above)
Comment (nickoola) [reply src/itemData/index.tsx]: I'm also asking myself about onClick, onTouchStart, onTouchEnd, onMouseDown, onMouseUp, onBlur and onFocus like in the Button component.

File: src/_utils/item/Item.tsx (hunk 1 above)
Comment (nickoola) [reply src/_utils/item/Item.tsx]: Finally removed

File: src/_utils/item/Item.tsx [hunk 4]
```diff
@@ -108,12 +110,14 @@ const Item = ({
       <div className="kirk-item-leftText">
         {leftTitle && (
           <Text
-            className={leftBody ? 'kirk-item-title' : null}
```
Comment (nickoola) [inline src/_utils/item/Item.tsx]: I think it's a legacy rule. There is no logic display link between leftBody inside leftTitle.
And kirk-item-title doesn't apply any styles.

File: src/_utils/item/Item.tsx [hunk 5]
```diff
@@ -18,6 +18,7 @@ export interface ItemProps {
   readonly highlighted?: boolean
   readonly isClickable?: boolean
   readonly leftTitle?: string
+  readonly leftTitleButtonAddon?: React.ReactElement<any>
```
Comment (mcellier) [inline src/_utils/item/Item.tsx]: In theory, not `any`, but `Button`?

File: src/_utils/item/Item.tsx [hunk 6]
```diff
@@ -108,12 +110,14 @@ const Item = ({
...
           <Text
-            className={leftBody ? 'kirk-item-title' : null}
+            className={leftTitleButtonAddon ? 'kirk-item-title--withButtonAddon' : null}
             display={leftTitleDisplay}
             textColor={leftTitleColor}
             tag={TextTagType.DIV}
           >
             {leftTitle}
+            {/* In case of a clickable Item don't display the info button */}
+            {!href && leftTitleButtonAddon && leftTitleButtonAddon}
```
Comment (mcellier) [inline src/_utils/item/Item.tsx]: Double `leftTitleButtonAddon `

File: src/_utils/item/Item.tsx (hunk 5 above)
Comment (nickoola) [reply src/_utils/item/Item.tsx]: Yes good catch thanks!

File: src/_utils/item/Item.tsx (hunk 6 above)
Comment (nickoola) [reply src/_utils/item/Item.tsx]: Fixed

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 
+- **[NEW]** Add info button props to `Item` and `ItemData`
```
Comment (mcellier) [inline CHANGELOG.md]: nit: Add `leftTitleButtonAddon` property to `ItemData`

Item isn't readlly exposed, so no need to talk about it here.

**Pull request merged**